ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,669 @@
+                            nullptr;
+
+  if (m_utilityAssetType)
+  {
+    const QList<UtilityTerminal*> utilityTerminals = m_utilityAssetType->terminalConfiguration()->terminals();
+
+    UtilityTerminal* loadTerminal = nullptr;
+
+    for (UtilityTerminal* utilityTerminal : utilityTerminals)
+    {
+      // Set the terminal for the location. (For our case, use the "Load" terminal.)
+      if (utilityTerminal->name() == "Load")
+      {
+        loadTerminal = utilityTerminal;
+        break;
+      }
+    }
+    m_startingLocation = m_utilityNetwork->createElementWithAssetType(m_utilityAssetType, QUuid("{1CAF7740-0BF4-4113-8DB2-654E18800028}"), loadTerminal, this);
+  }
+  else
+    return;
+
+  // Display starting location as graphic
+  m_utilityNetwork->featuresForElementsAsync(QList<UtilityElement*> {m_startingLocation}).then(this, [this](QList<ArcGISFeature*>)
+  {
+    ArcGISFeatureListModel* elementFeaturesList = m_utilityNetwork->featuresForElementsResult();
+    const Point startingLocationGeometry = geometry_cast<Point>(elementFeaturesList->first()->geometry());
+
+    GraphicsOverlay* overlay = new GraphicsOverlay(this);
+    SimpleMarkerSymbol* startingPointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Cross, Qt::green, 25, this);
+    Graphic* graphic = new Graphic(startingLocationGeometry, startingPointSymbol, this);
+    overlay->graphics()->append(graphic);
+
+    m_mapView->graphicsOverlays()->append(overlay);
+  });
+
+  UtilityNetworkDefinition* networkDefinition = m_utilityNetwork->definition();
+  UtilityDomainNetwork* domainNetwork = networkDefinition->domainNetwork("ElectricDistribution");
+  UtilityTier* tier = domainNetwork->tier("Medium Voltage Radial");
+
+  m_traceParameters = new UtilityTraceParameters(UtilityTraceType::Downstream, {m_startingLocation}, this);
+
+  UtilityTraceConfiguration* m_traceConfiguration = tier->defaultTraceConfiguration();
+
+  m_traceParameters->setTraceConfiguration(m_traceConfiguration);
+
+  m_isValidateBtnEnabled = m_utilityNetwork->definition()->capabilities()->isSupportsValidateNetworkTopology();
+  emit isValidateBtnEnabled();
+  m_isTraceBtnEnabled = m_utilityNetwork->definition()->capabilities()->isSupportsTrace();
+  emit isTraceBtnEnabled();
+  m_isStateBtnEnabled = m_utilityNetwork->definition()->capabilities()->isSupportsNetworkState();
+  emit isStateBtnEnabled();
+  m_isClearBtnEnabled = false;
+  emit isClearBtnEnabled();
+
+  updateMessage("Utility Network Loaded\n"
+      "Tap on a feature to edit.\n"
+      "Click 'Get State' to check if validating is\n"
+      "necessary or if tracing is available.\n"
+      "Click 'Trace' to run a trace.");
+
+  m_busy = false;
+  emit isBusy();
+}
+
+LabelDefinition* ValidateUtilityNetworkTopology::createDeviceLabelDefinition() const
+{
+  SimpleLabelExpression* labelExpression = new SimpleLabelExpression("[devicestatus]");
+
+  TextSymbol* textSymbol = new TextSymbol();
```

Comment:
missing parents - any time you use `new`, you know you'll need to specify a parent object

```suggestion
  SimpleLabelExpression* labelExpression = new SimpleLabelExpression("[devicestatus]", this);

  TextSymbol* textSymbol = new TextSymbol(this);
```